client/wfprev-war/src/main/angular/src/app/components/edit-project/fiscal-map/fiscal-map.component.ts

```diff
@@ -0,0 +1,318 @@
+          weight: 2,
+          fillOpacity: 0.1
+        }
+      };
+  
+      const addToMap = (geom: any) => {
+        L.geoJSON(geom, geoJsonOptions).addTo(this.map!);
+      };
+  
+      if (geometry.type === 'GeometryCollection') {
+        geometry.geometries.forEach((subGeom: any) => addToMap(subGeom));
+      } else {
+        addToMap(geometry);
+      }
+    });
+  }
+  
+
+  initMap(): void {
+    const mapContainer = document.getElementById('fiscalMap');
+    if (mapContainer && (mapContainer as any)._leaflet_id != null) {
+      (mapContainer as any)._leaflet_id = null;
+    }
+
+    this.map = L.map('fiscalMap', {
+      center: [49.00005, -124.0001], // Center of the boundary
+      zoom: 14,
+      zoomControl: true,
+    });
+
+    (this.map.zoomControl as L.Control.Zoom).setPosition('topright');
+
+    L.tileLayer('https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png', {
+      attribution: '© OpenStreetMap contributors'
+    }).addTo(this.map);
+
+    const legendHelper = new LeafletLegendService();
+    legendHelper.addLegend(this.map!, this.fiscalColorMap);
+    const bcBounds = L.latLngBounds([48.3, -139.1], [60.0, -114.0]);
+    this.map.fitBounds(bcBounds, { padding: [20, 20] });
+    createFullPageControl(() => this.openFullMap()).addTo(this.map!);
+  }
+
+  openFullMap(): void {
+    const latLngs: L.LatLng[] = [];
+    // handle this based on following steps and scenarios.
+    // 1. Add activity boundaries
+    this.allActivityBoundaries?.forEach(entry => {
+      entry.boundary?.forEach((item: any) => {
+        const geometry = item.geometry;
+        const layer = L.geoJSON(geometry);
+        const layerBounds = layer.getBounds();
+        latLngs.push(layerBounds.getSouthWest());
+        latLngs.push(layerBounds.getNorthEast());
+      });
+    });
+  
+    // 2. Add project boundaries
+    this.projectBoundary?.forEach((item: any) => {
+      const geometry = item.geometry;
+      const layer = L.geoJSON(geometry);
+      const layerBounds = layer.getBounds();
+      latLngs.push(layerBounds.getSouthWest());
+      latLngs.push(layerBounds.getNorthEast());
+    });
+  
+    let urlTree: UrlTree;
+  
+    // 3. If there are any polygons, use combined bounds
+    if (latLngs.length > 0) {
+      const bounds = L.latLngBounds(latLngs);
+      const bbox = [
+        bounds.getWest().toFixed(6),
+        bounds.getSouth().toFixed(6),
+        bounds.getEast().toFixed(6),
+        bounds.getNorth().toFixed(6),
+      ].join(',');
+  
+      urlTree = this.router.createUrlTree([ResourcesRoutes.MAP], {
+        queryParams: { bbox }
+      });
+  
+    } else if (this.projectLatitude && this.projectLongitude) {
+      // 4. No polygons, but project has coordinates → zoom to small area around point
+      const lat = parseFloat(this.projectLatitude);
+      const lng = parseFloat(this.projectLongitude);
+      const delta = 0.01;
+  
+      const bbox = [
+        (lng - delta).toFixed(6),
```

Comment:
Any specific reason for fixing to 6?

Reply:
Just concern about the coordinate has extremely long decimals.  But just confirmed when user add coordinate via Create Project, it is already being handled. So I will remove this.